public/js/index.js

```diff
@@ -1,5 +1,32 @@
-function displayMessage() {
```

Comment:
@arreche, In the `karma.conf.js`, all the files in `files` array gets run before the `index_spec.js`.  But when Karma runs, it does not load our `index.html`. `index.js` cannot find `<div class='app'></div>`.  So `document.getElementById('app')` returns null.  When you try to append to this null element it throws an error and karma errors out.

I see several ways to make our test work.  (**This is based on the commit before the workaround where your were appending to a div element not the body**)

1.  Create a `html.js` in `test/` that appends`<div class='app'></div>` to `document.body` and add it to `files` array in `karma.config.js` as the first element.  So that when `index.js` runs it `document.getElementById('app')` does not return null.

2. Make appending products list to `<div class='app'></div>` a function.  You can have something like this,

    ````
    function displayProducts(products) {
      var app = document.getElementById('app');
      app.appendChild(buildProducts(products));
    }
    ```
We would still have to create `<div class='app'></div>` in `index_spec.js` to test `displayProducts`.

3. We can separate functions and statements into different files.  We would need to move all the functions in `index.js` to something like `displayResults.js` and not include `index.js` in `files` array in `karma.config.js`.